vernac/egramrocq.ml

```diff
@@ -102,38 +101,33 @@ let create_pos = function
 let find_position_gen current ensure assoc lev =
   match lev with
   | None ->
-    current, (ReuseFirst, None, None, None)
+    current, (ReuseFirst, None, None)
   | Some n ->
-    let after = ref None in
-    let init = ref None in
-    let rec add_level q = function
-      | (p,_,_ as pa)::l when p > n -> pa :: add_level (Some p) l
-      | (p,a,reinit)::l when Int.equal p n ->
-        if reinit then
+    let rec add_level previous = function
+      | (p,_,_ as pa)::l when p > n ->
+        let updated, res = add_level (Some p) l in
+        pa :: updated, res
+      | (p,a,empty)::l when Int.equal p n ->
+        if empty then
+          (* XXX we ignore preexisting associativity for empty levels, is that what we want? *)
```

Comment:
@rocq-prover/parsing-maintainers does ignoring associativity for empty levels actually work without bugs? for instance
~~~coq
Print Grammar constr.
(* Entry term is
...
| "90" RIGHTA
  [  ]
...
*)

Notation "x -+- y" := (x + y) (left associativity, at level 90, only parsing).

Print Grammar constr.
(*
...
| "90" RIGHTA
  [ SELF; "-+-"; NEXT ]
...
*)

Set Printing Parentheses.
Check 0 -+- 1 -+- 2.
(* (0 + 1) + 2 *)
~~~
gramlib still thinks level 90 is RIGHTA but `-+-` gets parsed left associatively? What's the gramlib associativity used for then?